### ISSUE TITLE
Is there no use the pretrained dinov2 model ?

### ISSUE THREAD

**Issue by BarneetWei** [NVlabs/FoundationStereo#220]
class DPT_DINOv2(nn.Module):
    def __init__(self, encoder='vitl', features=256, out_channels=[256, 512, 1024, 1024], use_bn=False, use_clstoken=False, pretrained_dino=False):
        super(DPT_DINOv2, self).__init__()

        assert encoder in ['vits', 'vitb', 'vitl']

        # in case the Internet connection is not stable, please load the DINOv2 locally
        # if localhub:
        #     self.pretrained = torch.hub.load('torchhub/facebookresearch_dinov2_main', 'dinov2_{:}14'.format(encoder), source='local', pretrained=False)
        # else:
        self.pretrained = torch.hub.load('facebookresearch/dinov2', 'dinov2_{:}14'.format(encoder), pretrained=pretrained_dino)

        dim = self.pretrained.blocks[0].attn.qkv.in_features

        self.depth_head = DPTHead(1, dim, features, use_bn, out_channels=out_channels, use_clstoken=use_clstoken)

    def forward(self, x):
        h, w = x.shape[-2:]

        features = self.pretrained.get_intermediate_layers(x, 4, return_class_token=True)
        patch_size = self.pretrained.patch_size
        patch_h, patch_w = h // patch_size, w // patch_size
        output = self.depth_head(features, patch_h, patch_w, patch_size=patch_size, return_intermediate=True)
        return output

**Comment by wenbowen123** [comment]
Hi, we didn't use DINO features, but Depth-Anything V2.

**Comment by BarneetWei** [comment]
okay, thanks.
